src/main/java/org/apache/pulsar/io/jcloud/partitioner/TimePartitioner.java

```diff
@@ -59,16 +59,28 @@ private long parseDurationString(String timePartitionDuration) {
         if (StringUtils.isBlank(timePartitionDuration)) {
             return DEFAULT_PARTITION_DURATION;
         }
-        String number = timePartitionDuration.substring(0, timePartitionDuration.length() - 1);
-        switch (timePartitionDuration.charAt(timePartitionDuration.length() - 1)) {
-            case 'd':
-            case 'D':
-                return Long.parseLong(number) * 24L * 3600L * 1000L;
-            case 'h':
-            case 'H':
-                return Long.parseLong(number) * 3600L * 1000L;
-            default:
-                throw new RuntimeException("not support timePartitionPattern " + timePartitionDuration);
+        if (Character.isAlphabetic(timePartitionDuration.charAt(timePartitionDuration.length() - 1))) {
+            String number = timePartitionDuration.substring(0, timePartitionDuration.length() - 1);
+            switch (timePartitionDuration.charAt(timePartitionDuration.length() - 1)) {
+                case 'd':
+                case 'D':
+                    return Long.parseLong(number) * 24L * 3600L * 1000L;
+                case 'h':
+                case 'H':
+                    return Long.parseLong(number) * 3600L * 1000L;
+                case 'm':
```

Comment:
do you need upper case `M` and `S`?

Reply:
I was referred from https://momentjs.com/docs/#/durations/, which usually use the uppercase `M` for `month`, so I haven't put it to the switch block.